File: 2022/haskell/app/X1.hs
import Data.List.Split (splitOn)
import Data.Sort (sort)

parse = map (sum . map read . lines) . splitOn "\n\n"
partOne = foldr1 max
partTwo = sum . take 3 . reverse . sort

main :: IO ()
main = do
  input <- parse <$> readFile "../inputs/1"
  print (partOne input)
  print (partTwo input)

File: 2022/ruby/x1.rb
def parse
  file = File.open("../inputs/1", "r")
  contents = file.read
                 .split("\n\n")
                 .map { |x| x.lines }
                 .map { |x| x.map {|y| Integer(y) } }
                 .map { |x| x.sum }

  file.close
  return contents
end

if __FILE__ == $0
  puts parse.max
  puts parse.sort.reverse.take(3).sum
end
